street_network/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,128 @@
 use crate::osm::NodeID;
 use crate::IntersectionComplexity::*;
-use crate::{Direction, IntersectionComplexity, StreetNetwork};
+use crate::{
+    ConflictType, IntersectionComplexity, OriginalRoad, RestrictionType, Road, StreetNetwork,
+};
+use std::cmp::{max, min};
+use std::collections::BTreeMap;
 
 /// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
 /// considered "unclassified" and will be updated with a guess, others will be left unchanged.
 pub fn classify_intersections(streets: &mut StreetNetwork) {
-    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    let mut changes: Vec<(NodeID, (IntersectionComplexity, ConflictType))> = Vec::new();
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
             changes.push((*id, guess_complexity(streets, id)));
         }
     }
 
     for (id, complexity) in changes.into_iter() {
-        streets.intersections.get_mut(&id).unwrap().complexity = complexity;
+        let intersection = streets.intersections.get_mut(&id).unwrap();
+        intersection.complexity = complexity.0;
+        intersection.conflict_level = complexity.1;
     }
 }
 
 /// Guesses the complexity of the intersection based on the connecting roads and their lanes.
 ///
 /// The existing complexity field is ignored, so be careful how you use the guessed value.
-fn guess_complexity(streets: &StreetNetwork, intersection_id: &NodeID) -> IntersectionComplexity {
-    let roads = streets.roads_per_intersection(*intersection_id);
+fn guess_complexity(
+    streets: &StreetNetwork,
+    intersection_id: &NodeID,
+) -> (IntersectionComplexity, ConflictType) {
+    use ConflictType::*;
+    let road_ids = streets.roads_per_intersection(*intersection_id);
+    let roads: Vec<&Road> = road_ids
+        .iter()
+        .map(|id| streets.roads.get(id).unwrap())
```

Comment:
For a btreemap, we can just do `&streets.roads[id]` equivalently, which'll crash if the key isn't there